### PR TITLE
[MCC-575482] Add more configuration options

### PR DESCRIPTION
**Pull request by cmcinnes-mdsol**
@mdsol/team-16 may need to change this MCC to a separate story but I wanted to make the fallback to v1 when v2 fails configurable. I also thought adding a `v1_only_sign_requests` flag could ease rollout of the improved MWSV2 protocol (with double encoding removed) so that we don't have to have all ruby services on mauth client < v5.1.0 to upgrade twice. 

Adding DNM because this is pointed at the branch from #39

### CODE REVIEW

File: doc/mauth.yml.md [hunk 1]
```diff
@@ -66,8 +69,10 @@ test:
 - `app_uuid` - Required in the same circumstances where a `private_key` is required.
 - `mauth_baseurl` - Required for authentication but not for signing. Needed for local authentication to retrieve public keys and for remote authentication. Usually this is `https://mauth.imedidata.com` for production.
 - `mauth_api_version` - Required for authentication but not for signing. only `v1` exists as of this writing.
-- `v2_only_authenticate` - If true, all outgoing requests will be signed with only the V2 protocol.
-- `v2_only_sign_requests` - If true, any incoming request or incoming response that does not use the V2 protocol will be rejected.
+- `v2_only_sign_requests` - If true, all outgoing requests will be signed with only the V2 protocol.
+- `v2_only_authenticate` - If true, any incoming request or incoming response that does not use the V2 protocol will be rejected.
+- `disable_fallback_to_v1_on_v2_failure` - If true, any incoming V2 requests that fail authentication will fall back to V1 authentication.
```
Comment (ejennings-mdsol) [inline doc/mauth.yml.md:74]: The flag is called `disable_fallback_to_v1_on_v2_failure` but you say if true, i.e. fallback is disabled, then it sounds like v2 authentication failures should **not** fallback if true.

File: spec/support/shared_examples/authenticator_base.rb [hunk 2]
```diff
@@ -1,3 +1,4 @@
+require 'byebug'
```
Comment (ejennings-mdsol) [inline spec/support/shared_examples/authenticator_base.rb:1]: Can this be removed?

File: doc/mauth.yml.md (hunk 1 above)
Comment (cmcinnes-mdsol) [reply doc/mauth.yml.md:74]: Ah yes thanks, wrote that description before i switched it from `enable` to `disable` fallback.

**Comment by cmcinnes-mdsol** [comment]
@ejennings-mdsol with `disable_fallback_to_v1_on_v2_failure` set to true the client can still authenticate requests that are only signed with V1. 
With `v2_only_authenticate` set to true all requests signed with only V1 will be considered inauthentic.

**Comment by ykitamura-mdsol** [comment]
I'm just throwing an idea, might not work well 😅 but I thought about what will happen when we have a new version `v3` and the configuration could be more complicated.

Maybe we could just set versions, something like:
```yaml
authenticate_versions: v1,v2
sign_versions: v1,v2
```

Also in the [`mauth.yml.dice` file](https://github.com/mdsol/mauth-client-ruby/pull/40/files#diff-99c74238cb6bed5ebf1a5ce524b2cfb9), probably we should add a prefix `mauth_` to clearly differentiate from others so this will look like:
```yaml
common: &common
  mauth_baseurl: <%= configured.mauth_url! || 'http://localhost:7000' %>
  mauth_api_version: v1
  app_uuid: <%= configured.mauth_app_uuid! || 'fb17460e-9868-11e1-8399-0090f5ccb4d3' %>
  private_key_file: config/mauth_key
  authenticate_versions: <%= configured.mauth_authenticate_versions || 'v1,v2' %>
  sign_versions: <%= configured.mauth_sign_versions || 'v1,v2' %>
```

**Comment by cmcinnes-mdsol** [comment]
I like it @ykitamura-mdsol 👍

**Comment by cmcinnes-mdsol** [comment]
On second thought @ykitamura-mdsol that would require a lot of spec changes, and we don't want to let them authenticate with only V1, so I think for V3 maybe we just cross that bridge when we come to it. Let me know if you feel strongly.

**Comment by ykitamura-mdsol** [comment]
> require a lot of spec changes

Oh, this is not good... I'm fine with kicking the can 😅

**Pull request merged**